test/src/test/java/org/corfudb/integration/StreamAddressDiscoveryIT.java

```diff
@@ -1002,4 +1002,78 @@ public void testStreamRebuiltWithHoleAsFirstEntryAfterTrimNoCP() throws Exceptio
             shutdownCorfuServer(server);
         }
     }
+
+    /**
+     * Test rebuilding a stream from a new runtime, whenever the last address before the checkpoint
+     * was a hole.
+     *
+     * This case is interesting to test as addresses that become holes are discarded by the streamView,
+     * while sequencer's are agnostic of this info, hence, take it into account for trim mark computation.
+     */
+    @Test
+    public void testCheckpointWhenLastAddressIsHole() throws Exception {
+        final int numEntries = 10;
```

Comment:
Do we need a test that creates an empty checkpoint and restarts the server to verify that the sequencer bootstrap state is correct?

Reply:
Sure, this is a good idea, cause the test we have does not exercise reboot. Adding it.